server/lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -2,30 +2,138 @@ defmodule RealtimeWeb.RealtimeChannel do
+      {:ok, assign(socket, %{access_token: access_token, verify_ref: ref})}
+    else
+      _ -> {:error, %{reason: "error occurred when joining #{topic}"}}
+    end
   end
 
-  def handle_info(:after_join, socket) do
-    Realtime.Metrics.SocketMonitor.track_channel(socket)
-    {:noreply, socket}
+  def join("realtime:" <> _, _, socket) do
+    SocketMonitor.track_channel(socket)
+    {:ok, socket}
+  end
+
+  def handle_info(
+        :after_join,
+        %Socket{
+          assigns: %{access_token: access_token},
+          pubsub_server: pubsub_server,
+          serializer: serializer,
+          topic: topic,
+          transport_pid: transport_pid
+        } = socket
+      ) do
+    with {:ok, %{"sub" => user_id}} <- ChannelsAuthorization.authorize(access_token),
+         {:ok, bin_user_id} <- Ecto.UUID.dump(user_id) do
+      :ok = PubSub.unsubscribe(pubsub_server, topic)
+      user_fastlane = {:user_fastlane, transport_pid, serializer, bin_user_id}
+      :ok = PubSub.subscribe(pubsub_server, topic, metadata: user_fastlane)
+      {:noreply, socket}
+    else
+      _ -> {:stop, :invalid_access_token, socket}
+    end
+  end
+
+  def handle_info(
+        %Broadcast{
+          event: "sync_subscription",
+          topic: "subscription_manager"
+        },
+        %Socket{
+          assigns: %{access_token: access_token},
+          channel_pid: channel_pid,
+          topic: "realtime:" <> subtopic
+        } = socket
+      ) do
+    with {:ok, %{"sub" => user_id, "email" => email}} <-
+           ChannelsAuthorization.authorize(access_token),
+         {:ok, bin_user_id} <- Ecto.UUID.dump(user_id),
+         :ok <-
+           SubscriptionManager.track_topic_subscriber(%{
+             channel_pid: channel_pid,
+             topic: subtopic,
+             user_id: bin_user_id,
+             email: email
+           }) do
+      {:noreply, socket}
+    else
+      _ -> {:stop, :sync_subscription_error, socket}
+    end
+  end
+
+  def handle_info(
+        :verify_access_token,
+        %Socket{
+          assigns: %{access_token: access_token, verify_ref: ref}
+        } = socket
+      ) do
+    Process.cancel_timer(ref)
+
+    case ChannelsAuthorization.authorize(access_token) do
+      {:ok, _} ->
+        ref = Process.send_after(self(), :verify_access_token, @verify_token_interval)
+        {:noreply, assign(socket, :verify_ref, ref)}
+
+      :error ->
+        {:stop, :invalid_access_token, socket}
+    end
+  end
+
+  def handle_in(
+        "access_token",
+        %{"access_token" => fresh_token},
+        %Socket{
+          assigns: %{access_token: _},
+          channel_pid: channel_pid,
+          topic: "realtime:" <> subtopic
+        } = socket
+      ) do
+    with {:ok, %{"sub" => user_id, "email" => email}} <-
+           ChannelsAuthorization.authorize(fresh_token),
+         {:ok, bin_user_id} <- Ecto.UUID.dump(user_id),
+         :ok <-
+           SubscriptionManager.track_topic_subscriber(%{
```

Comment:
In `SubscriptionManager.track_topic_subscriber/1` need to check if the current process is already monitored. For now, it always add monitoring

Reply:
@abc3 

> check if the current process is already monitored

how do i check if process is already monitored?

Reply:
yeah, just check the state